File: TAREA CRUD DEPENDIENTE SPRIN BOT/AngularFRONT/SiSCMSUPeUAngular2022-main/src/app/producto.ts
// export enum Status {
//   ACTIVO = '1',
//   INACTIVO = '0'
// }
export class Producto {
  id?:number;
  nombre?:string;
  descripccion?:string;
  detalles?:string;
  precio?: number | null | undefined | string;
  // Precio?:number;
  oferta?:string;
  color?:string;
  talla?:string;
  categorias_Id?: string;
  // status?: Status;
  // categoriasId?: number;


}

File: TAREA CRUD DEPENDIENTE SPRIN BOT/SprinBotBACKEN/catalogo/catalogo/src/main/java/com/example/catalogo/entity/Producto.java
package com.example.catalogo.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "producto")
public class Producto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer Id;
    private String Nombre;
    private String Descripccion;
    private String Detalles;
    private Double Precio;
    private String Oferta;
    private String Color;
    private String Talla;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "Categorias_id")
    private Categorias categorias;
    // @ManyToOne(fetch = FetchType.LAZY)
    // @JoinColumn(name = "Categorias_id")
    // private Categorias categorias;

}
